plugins/modules/fabric_devices_workflow_manager.py

```diff
@@ -0,0 +1,4080 @@
+                                      item.get("interfaceName") == interface_name)
+            l2_handoff_details = get_dict_result(all_l2_handoff_details, "internalVlanId", internal_vlan_id)
+            if l2_handoff_details:
+                self.log(
+                    "The L2 handoff details with the internal VLAN Id: {details}"
+                    .format(details=internal_vlan_id)
+                )
+                l2_handoff_id = l2_handoff_details.get("id")
+                break
+
+            end_time = time.time()
+            if (end_time - start_time) >= self.max_timeout:
+                self.msg = (
+                    "Max timeout of {max_time} sec has reached for the API 'l2_handoff_exists' status."
+                    .format(max_time=self.max_timeout)
+                )
+                self.status = "failed"
+                return self.check_return_status()
+
+        return l2_handoff_id
+
+    def sda_l3_handoff_exists(self, fabric_id, device_id, transit_name):
+        """
+        Check the availability of the L2 Handoff for the 'fabric_id', 'network_device_id',
+        'internal_vlan_id'.
+
+        Parameters:
+            fabric_id (str): The Id of the fabric site to check for existence.
+            device_id (str): The Id of the provisioned device to check for existence.
+            transit_name (str): The IP address of the the provisioned device.
+        Returns:
+            sda_l3_handoff_details (dict or None): The details of the L3 Handoff with SDA transit with
+            the given transit name. None if the there is no L3 Handoff with SDA transit.
+        Description:
+            Call the API 'get_fabric_devices_layer3_handoffs_with_sda_transit'. If the response is
+            empty return None. Else, return the details of the l3 handoff with SDA transit.
+        """
+
+        sda_l3_handoff_details = None
+        transit_id = self.get_transit_id_from_name(transit_name)
+        if not transit_id:
+            self.msg = (
+                "The SDA transit with the name '{name}' is not available in the Cisco Catalyst Center."
+                .format(name=transit_name)
+            )
+            self.status = "failed"
+            return self.check_return_status()
+
+        start_time = time.time()
+        offset = 1
+
+        # Call the SDK with incremental offset till to find the SDA L3 Handoff
+        while True:
+            try:
+                all_sda_l3_handoff_details = self.dnac._exec(
+                    family="sda",
+                    function="get_fabric_devices_layer3_handoffs_with_sda_transit",
+                    op_modifies=True,
+                    params={
+                        "fabric_id": fabric_id,
+                        "network_device_id": device_id,
+                        "offset": offset
+                    }
+                )
+            except Exception as msg:
```

Comment:
Received API response from 'get_fabric_devices_layer3_handoffs_with_sda_transit': {}

We can move the exception before returning...